tools/setup.py

```diff
@@ -137,6 +132,17 @@ def generate_gn_args(mode):
     if rustc_wrapper:
         out += ['rustc_wrapper=%s' % gn_string(rustc_wrapper)]
 
+    # On Windows, we need to use a custom toolchain, for the following reasons:
+    #   * When building a static library, the default chromium toolchain
+    #     produces 'thin archives' when building a static library. These are not
+    #     compatible with Microsoft's link.exe, and therefore rustc can't use
+    #     such libraries out of the box.
+    #   * The default toolchain doesn't support cc_wrapper.
+    # TODO: Is there a way to use it without the involvement of args.gn?
+    if os.name == "nt":
+        tc = "//build_extra/toolchain/win:win_clang_x64"
+        out += ['custom_toolchain="%s"' % tc, 'host_toolchain="%s"' % tc]
```

Comment:
Configuration shouldn't be in setup.py - configuration should be in gn files - that's the purpose. I should have said this before when the above configurations were added here.
 We should fork `BUILDCONFIG.gn` and set the toolchain other settings in this file there. Alternatively you can start `tools/args_win.gn` and copy that into place.

Reply:
Sorry, not possible without forking //build.

Reply:
can you explain? can't you just change this line?
https://github.com/denoland/deno/blob/master/.gn#L6